zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/client/DedicatedZkClient.java

```diff
@@ -217,12 +279,23 @@ public String create(String path, Object data, CreateMode mode) {
     return create(path, data, ZooDefs.Ids.OPEN_ACL_UNSAFE, mode);
   }
 
+  @Override
+  public String create(String path, Object data, CreateMode mode, long ttl) {
+    return create(path, data, ZooDefs.Ids.OPEN_ACL_UNSAFE, mode, ttl);
+  }
+
   @Override
   public String create(String path, Object datat, List<ACL> acl, CreateMode mode) {
```

Comment:
Another question: how do you prevent newer ZK client talking to older version ZK server and calling the above APIs?

Reply:
This is a good question, we can't really prevent them because we have no way of knowing the ZK's version, but ZK itself (whether older version or newer version without the extended types enabled as explained [here](https://zookeeper.apache.org/doc/current/zookeeperAdmin.html#sc_configuration)) will throw a `KeeperException.UnimplementedException`.